modules/cfe_testcase/src/es_info_test.c

```diff
@@ -0,0 +1,263 @@
+    const char *     ESAppName   = "CFE_ES";
+    const char *     InvalidName = "INVALID_NAME";
+    char             AppNameBuf[OS_MAX_API_NAME + 4];
+    CFE_ES_AppInfo_t TestAppInfo;
+    CFE_ES_AppInfo_t ESAppInfo;
+
+    UtPrintf("Testing: CFE_ES_GetAppID, CFE_ES_GetAppIDByName, CFE_ES_GetAppName, CFE_ES_GetAppInfo");
+
+    UtAssert_INT32_EQ(CFE_ES_GetAppIDByName(&AppIdByName, TestAppName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetAppID(&TestAppId), CFE_SUCCESS);
+    UtAssert_ResourceID_EQ(TestAppId, AppIdByName);
+    UtAssert_INT32_EQ(CFE_ES_GetAppName(AppNameBuf, TestAppId, sizeof(AppNameBuf)), CFE_SUCCESS);
+    UtAssert_StrCmp(AppNameBuf, TestAppName, "CFE_ES_GetAppName() = %s", AppNameBuf);
+
+    UtAssert_INT32_EQ(CFE_ES_GetAppInfo(&TestAppInfo, TestAppId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetAppIDByName(&ESAppId, ESAppName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ES_GetAppInfo(&ESAppInfo, ESAppId), CFE_SUCCESS);
+
+    UtAssert_True(TestAppInfo.Type == CFE_ES_AppType_EXTERNAL, "Test App Info -> Type = %d", (int)TestAppInfo.Type);
+    UtAssert_True(ESAppInfo.Type == CFE_ES_AppType_CORE, "ES App Info -> Type = %d", (int)ESAppInfo.Type);
+
+    UtAssert_StrCmp(TestAppInfo.Name, TestAppName, "Test App Info -> Name = %s", TestAppInfo.Name);
+    UtAssert_StrCmp(ESAppInfo.Name, ESAppName, "ES App Info -> Name = %s", ESAppInfo.Name);
+
+    UtAssert_StrCmp(TestAppInfo.EntryPoint, "CFE_TR_AppMain", "Test App Info -> EntryPt  = %s", TestAppInfo.EntryPoint);
+    UtAssert_True(strlen(ESAppInfo.EntryPoint) == 0, "ES App Info -> EntryPt  = %s", ESAppInfo.EntryPoint);
+
+    UtAssert_StrCmp(TestAppInfo.FileName, "/cf/cfe_testrunner.so", "Test App Info -> FileName = %s",
+                    TestAppInfo.FileName);
+    UtAssert_True(strlen(ESAppInfo.FileName) == 0, "ES App Info -> FileName  = %s", ESAppInfo.FileName);
+
+    UtAssert_True(TestAppInfo.StackSize > 0, "Test App Info -> StackSz  = %d", (int)TestAppInfo.StackSize);
+    UtAssert_True(ESAppInfo.StackSize > 0, "ES App Info -> StackSz  = %d", (int)ESAppInfo.StackSize);
+
+    if (TestAppInfo.AddressesAreValid)
+    {
+        UtAssert_True(TestAppInfo.AddressesAreValid > 0, "Test App Info -> AddrsValid?  = %d",
+                      (int)TestAppInfo.AddressesAreValid);
+        UtAssert_True(TestAppInfo.CodeAddress > 0, "Test App Info -> CodeAddress  = %d", (int)TestAppInfo.CodeAddress);
+        UtAssert_True(TestAppInfo.CodeSize > 0, "Test App Info -> CodeSize  = %d", (int)TestAppInfo.CodeSize);
+        UtAssert_True(TestAppInfo.DataAddress > 0, "Test App Info -> DataAddress  = %d", (int)TestAppInfo.DataAddress);
+        UtAssert_True(TestAppInfo.DataSize > 0, "Test App Info -> DataSize  = %d", (int)TestAppInfo.DataSize);
+        UtAssert_True(TestAppInfo.BSSAddress > 0, "Test App Info -> BSSAddress  = %d", (int)TestAppInfo.BSSAddress);
+        UtAssert_True(TestAppInfo.BSSSize > 0, "Test App Info -> BSSSize  = %d", (int)TestAppInfo.BSSSize);
```

Comment:
addresses should not be cast as `int`, they should be `unsigned long`

Reply:
fixed